### ISSUE TITLE
Call Web API from Angular with Azure AD: audience exception

### ISSUE THREAD

**Issue by daviddejaeger** [AzureAD/microsoft-authentication-library-for-js#1797]
Please follow the issue template below. Failure to do so will result in a delay in answering your question.

## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [ ] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [x] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description
Hi, 

I'm not sure if this is a bug or I'm doing something wrong but when I call my ASP.NET Core Web API from my Angular Application I got the following error on my server:
"Microsoft.IdentityModel.Tokens.SecurityTokenInvalidAudienceException: IDX10214: Audience validation failed".

I've logged the access token in my Angular application and decoded it. I also have another Blazor WA application using this backend (also with authorization and there is works). I have compared the 2 decoded tokens and noticed a difference in the 'aud' property:

For my 'msal-angular' token this is:
{
  "aud": "api://{cliend-id}",
   ...
}
Where as in my Blazor application this is:
{
  "aud": "{cliend-id}",
   ...
}

I suspect this "api://" prefix in the audience property is the problem but I don't know where this is coming from. I checked both client applications in the app registration in azure AD and they are the same. I have not found the solution or answer why this is happening yet.

**Comment by daviddejaeger** [comment]
Update:

I have found the solution. It turns out I was using the server side from the blazor template which was using:

`services.AddAuthentication(AzureADDefaults.BearerAuthenticationScheme)
                .AddAzureADBearer(options => Configuration.Bind("AzureAd", options));`

However I noticed the template from msal-angular was using:
`services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
                .AddProtectedWebApi(Configuration);`

This comes from microsoft.identity.web which is in preview. Inside the method _AddProtectedWebApi_  is the code that handles this:

`// The valid audiences are both the Client ID (options.Audience) and api://{ClientID}
                options.TokenValidationParameters.ValidAudiences = new string[]
                {
                    options.Audience, $"api://{options.Audience}"
                };`

This can be closed.
